clusterctl/clusterdeployer/clusterdeployer_test.go

```diff
@@ -484,6 +483,72 @@ func TestCreateProviderComponentsScenarios(t *testing.T) {
 	}
 }
 
+func TestExtractMasterMachine(t *testing.T) {
+	const singleMasterName = "test-master"
+	multpleMasterNames := []string{"test-master-1", "test-master-2"}
+	const singleNodeName = "test-node"
+	multipleNodeNames := []string{"test-node-1", "test-node-2", "test-node-3"}
+
+	testCases := []struct {
+		name            string
+		inputMachines   []*clusterv1.Machine
+		expectedMasters *clusterv1.Machine
+		expectedNodes   []*clusterv1.Machine
+		expectedError   error
+	}{
+		{
+			name:            "success_1_master_1_node",
+			inputMachines:   generateMachines(),
+			expectedMasters: generateMasterNode(singleMasterName),
+			expectedNodes:   generateTestNodeMachines([]string{singleNodeName}),
+			expectedError:   nil,
+		},
+		{
+			name:            "success_1_master_multiple_nodes",
+			inputMachines:   generateValidExtractMasterMachineInput(singleMasterName, multipleNodeNames),
+			expectedMasters: generateMasterNode("test-master"),
```

Comment:
nit: pass singleMasterName here instead of the raw string.

Reply:
I had missed that. Thanks!